public/app/features/templating/variable_srv.ts

```diff
@@ -286,6 +291,22 @@ export class VariableSrv {
     return this.variableUpdated(variable);
   }
 
+  templateVarsChangedInUrl(vars: UrlQueryMap) {
+    const update: Array<Promise<any>> = [];
+    this.variables.forEach(v => {
+      const key = `var-${v.name}`;
+      if (vars.hasOwnProperty(key)) {
+        update.push(v.setValueFromUrl(vars[key]));
```

Comment:
@torkelo -- is there any way to know if this actually makes a change?

This function should only be called when the URL parameters change... but depending on how it is triggered I think that can happen twice.

Reply:
> @torkelo -- is there any way to know if this actually makes a change?
> This function should only be called when the URL parameters change... but depending on how it is triggered I think that can happen twice.

not sure what you mean